flink-tests/src/test/java/org/apache/flink/test/checkpointing/ZooKeeperHighAvailabilityITCase.java

```diff
@@ -187,6 +187,7 @@ public void testRestoreBehaviourWithFaultyStateHandles() throws Exception {
 		env.setParallelism(1);
 		env.setRestartStrategy(RestartStrategies.fixedDelayRestart(Integer.MAX_VALUE, 0));
 		env.enableCheckpointing(10); // Flink doesn't allow lower than 10 ms
+		env.getCheckpointConfig().setTolerableCheckpointFailureNumber(Integer.MAX_VALUE);
```

Comment:
Hm, this raises a bit of a red flag: if you need to adjust the test, does it mean that the user facing default behaviour is changed by the PR? What I mean, if we need to configure this to pass the tests do user have to change their job as well to get the old behaviour? This looks like either i) a failer that was tolerated before now fails the job or ii) the default should be `Integer.MAX_VALUE` and not `0`. Do you know which case applies?

Reply:
I think the default value should be `0` to keep the same behavior with `failOnCheckpointingErrors`(default is `true`). I changed it to fix the test case. Because of some test case (integration test) is not stable, sometimes the checkpoint would be declined, and they do not accept job recovery by `CheckpointFailureManager`, some of them do not allow restart, some of them will check the log files.

Reply:
My concern is: if the test was passing before then some user facing behaviour was probably changed and I want to understand why and if that is true we need to address it in some way.

Reply:
In total there are 4 such adjustments to tests. Also in many unit tests we pass in a NoOp `FailJobCallback`. Wonder if this also supresses some otherwise visible failing.

Reply:
Ok, let me find the specific old logs why it was failed and we can discuss based on it?

Reply:
I understand the default for `failOnCheckpointingErrors` was true and should stay like that, I want to make sure that now there are just not accidentally more cases that cause a fail than before. Because that is exactly what your change would surpress.

Reply:
Ok, sounds good. But you understand my concern, why it is something we need to investigate?

Reply:
I understand, my PR changed these test cases' default behavior and caused them should specify the definite tolerance number (not default). So they may trigger compatibility problem or cause more test cases failure.

Reply:
Log detail for this one: https://travis-ci.org/apache/flink/jobs/533358204

Reply:
All I get from the log is that it looks more like a deadlock, this might be unrelated to your changes and more about some changes to the network stack. Then the "fix" was probably just only a more lucky run. But that would be actually good news. I am wondering if that also applies for the other cases where you had to set the parameter to pass (e.g. `StatefulStreamingJob`)? Then I would double check if about the deadlock and you could revert to test changes. If this is all true, it is not the PRs fault be I still like to be on the safe side before we merge.

Reply:
Would be interesting to see the detailed log, but I cannot see the download link in the Travis log.

Reply:
Yes, actually, I am not very sure about this case.

Reply:
The bad thing is that I cannot access the jm/tm logs